docs/endpoints/multi_user.rst

```diff
@@ -0,0 +1,909 @@
+         "worker_init": { "type": "string" }
+      },
+      "additionalProperties": true
+   }
+
+Configuring a JSON document schema is out of scope for this documentation, but this tool
+is available to restrict what the MEP will accept from users for interpolation.  Please
+consult the `JSON Schema documentation <https://json-schema.org/>`_ for more information.
+
+``user_environment.yaml``
+-------------------------
+
+Use this file to specify site-specific environment variables to export to the UEP
+process.  Though this is a YAML file, it is interpreted internally as a simple
+top-level-only set of key-value pairs.  Nesting of data structures will probably not
+behave as expected.  Example:
+
+.. code-block:: yaml
+
+   SITE_SPECIFIC_VAR: --additional_flag_for_frobnicator
+
+That will be injected into the UEP process as an environment variable.
+
+
+Running the MEP
+===============
+
+The MEP starts in the exact same way as the CEP |nbsp| -- |nbsp| with the ``start``
+subcommand.  Unlike the CEP, however, the MEP has no notion of the ``detach_endpoint``
+configuration item.  Once started, the MEP stays attached to the console, with a timer
+that updates every second:
+
+.. code-block:: text
+
+    globus-compute-endpoint start debug_queue
+        >>> Multi-User Endpoint ID: [endpoint_uuid] <<<
+    ----> Fri Apr 19 11:56:27 2024
+
+The timer is only displayed if the agent is connected to the terminal, and is intended
+as a hint to the administrator that the agent is alive and working, even if no start
+UEP requests are yet incoming.
+
+And that's it.  The Multi-user endpoint is running, waiting for start UEP requests to
+come in.
+
+To stop the MEP, type ``Ctrl+\`` (SIGQUIT) or ``Ctrl+C`` (SIGINT).  Alternatively, the
+process also responds to SIGTERM.
+
+Checking the Logs
+-----------------
+
+If actively debugging or iterating, the two command line arguments ``--log-to-console``
+and ``--debug`` may be helpful as they increase the verbosity and color of the text to
+the console.  Meanwhile, the log is always available at
+``.globus_compute/<mt_endpoint_name>/endpoint.log``, and is the first place to look
+upon an unexpected behavior.  In a healthy MEP setup, there will be lots of lines about
+processes starting and stopping:
+
+.. code-block:: text
+
+   [...] Creating new user endpoint (pid: 3867325) [(harper, uep.4ade2ce0-9c00-4d8c-b996-4dff8fbb4bd0.e9097f8f-dcfc-3bc0-1b42-0b4ad5e3922a) globus-compute-endpoint start uep.4ade2ce0-9c00-4d8c-b996-4dff8fbb4bd0.e9097f8f-dcfc-3bc0-1b42-0b4ad5e3922a --die-with-parent]
+   [...] Command process successfully forked for 'harper' (Globus effective identity: b072d17b-08fd-4ada-8949-1fddca189b5e).
+   [...] Command stopped normally (3867325) [(harper, uep.4ade2ce0-9c00-4d8c-b996-4dff8fbb4bd0.e9097f8f-dcfc-3bc0-1b42-0b4ad5e3922a) globus-compute-endpoint start uep.4ade2ce0-9c00-4d8c-b996-4dff8fbb4bd0.e9097f8f-dcfc-3bc0-1b42-0b4ad5e3922a --die-with-parent]
+
+
+Advanced Environment Customization
```

Comment:
Since this is only intended for super users, whom likely already understand how to accomplish what is described, I think we should drop the section to hide the gun from non-super users.

Reply:
Per discussion, leaving this in.  The shim is a very important point for administrators to have as an option.

In discussion, the part that caught attention was this:

> While we don't suggest modifying this wrapper 

But the larger sentence is:

> While we don't suggest modifying this wrapper (for ease of future maintenance), one might inject another wrapper into the process, by modifying the process PATH and writing a custom `globus-compute-endpoint` wrapper:

And that's pretty clear I think.  "Don't overwrite this, _use_ this.  Inject your own."